### PR TITLE
Week 3 development step 3

### PR DESCRIPTION
**Pull request by keiffer01**
This step begins making a comment box for the portfolio using JSON as instructed from the walkthroughs. I created a new page called comments.html, with the navbar modified to reflect this. I also created a new java file CommentsServlet.java to handle server requests related to comments, and renamed the old DataServlet.java file to FactsServlet.java, which now solely handles serving fun facts. This branch just stores hard-coded ArrayList<String> data in the CommentsServlet.java file and returns it as JSON data when it receives a GET request. The comments.html file then displays the data.

Picture testing this with a hard coded array containing two elements "Hello" and "World":
Screenshot 2020-06-02 at 7 41 13 PM

TESTING: tested that comments.html was able to grab the array data from CommentsServlet.java and display it properly, with differing number of elements in the array.

### CODE REVIEW

File: portfolio/src/main/java/com/google/sps/servlets/CommentsServlet.java [hunk 1]
```diff
@@ -14,19 +14,38 @@
...
+  }
+
+  /* Returns comments as a JSON string. */
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    response.setContentType("text/html;");
-    response.getWriter().println("<h1>Hello world!</h1>");
+    // Convert comments to JSON using Gson
+    Gson gson = new Gson();
+    String json = gson.toJson(comments);
```
Comment (beiebi0) [inline portfolio/src/main/java/com/google/sps/servlets/CommentsServlet.java]: better naming? like commentsInJsonFormat?

Comment (keiffer01) [reply portfolio/src/main/java/com/google/sps/servlets/CommentsServlet.java]: Ok I've renamed the variable.

**Comment by keiffer01** [comment]
I also added error handling for the getComments function as reflected from the 'week 3 development step 2' pull request.

File: portfolio/src/main/java/com/google/sps/servlets/FactsServlet.java [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/** Servlet that returns fun facts. */
+@WebServlet("/facts")
+public class FactsServlet extends HttpServlet {
+
+  /* List of fun facts */
```
Comment (beiebi0) [inline portfolio/src/main/java/com/google/sps/servlets/FactsServlet.java]: nit: // should be fixed if other PR merges right?

**Review by beiebi0 — APPROVED** [review]
lgtm for the comment part.

File: portfolio/src/main/webapp/script.js [hunk 3]
```diff
@@ -13,30 +13,40 @@
 // limitations under the License.
 
 /*
- * Returns a random fun fact about me until all facts are exhausted.
+ * Returns a random fun fact about me from the /facts servlet.
```
Comment (HiramSilvey) [inline portfolio/src/main/webapp/script.js:16]: Remove extraneous "/"?

File: portfolio/src/main/webapp/script.js [hunk 4]
```diff
@@ -13,30 +13,40 @@
...
+    document.getElementById('facts').innerText = fact;
+  });
+}
+
+/*
+ * Returns comments from the /comments servlet.
+ *
+ * REQUIRES: none
+ * ENSURES: The 'comments-container' div container is replaced with the comments
+ *          stored in the /comments servlet as a comma-separated string. If the
```
Comment (HiramSilvey) [inline portfolio/src/main/webapp/script.js]: Unfinished sentence "If the"?

**Review by HiramSilvey — CHANGES_REQUESTED** [review]

File: portfolio/src/main/webapp/script.js (hunk 3 above)
Comment (keiffer01) [reply portfolio/src/main/webapp/script.js:16]: I put the slash there to indicate that it's from a servlet, but I'll put it in quotes to make it more obvious.

File: portfolio/src/main/webapp/script.js (hunk 4 above)
Comment (keiffer01) [reply portfolio/src/main/webapp/script.js]: Huh I think I accidentally deleted that line. It's been restored.

File: portfolio/src/main/java/com/google/sps/servlets/FactsServlet.java (hunk 2 above)
Comment (keiffer01) [reply portfolio/src/main/java/com/google/sps/servlets/FactsServlet.java]: Yeah I'll fix it with the other PR.

File: portfolio/src/main/webapp/script.js [hunk 5]
```diff
@@ -30,14 +30,15 @@ function getRandomFact() {
...
+ *          stored in the "/comments" servlet as a comma-separated string. If the
+ *          server returns an error, a nondescript error message is displayed instead.
  */
 function getComments() {
   fetch('/comments')
     .then(handleFetchErrors)
     .then(response => response.json())
-    .then(comments => {
+    .then(commentsInJson => {
       document.getElementById('comments-container').innerText = comments.toString();
```
Comment (HiramSilvey) [inline portfolio/src/main/webapp/script.js]: Does this line need to be updated to use "commentsInJson" as well?

**Review by HiramSilvey — CHANGES_REQUESTED** [review]

File: portfolio/src/main/webapp/script.js (hunk 5 above)
Comment (keiffer01) [reply portfolio/src/main/webapp/script.js]: Oh whoops, thanks for the catch.

**Review by HiramSilvey — APPROVED** [review]

**Pull request merged**